CRT.c

```diff
@@ -1001,6 +1001,26 @@ void CRT_setMouse(bool enabled) {
 }
 #endif
 
+static bool termKeySupport(const char* termType) {
```

Comment:
```c
static bool termSupportsDefinedKeys(const char* termType) {
   if (!termType)
      return false;

   if (termType[0] == 's') {
      if (termType[1] == 't' && (termType[2] == '-' || !termType[2])) {
         // Suckless Terminal. Regex "st(-)?".
         // Don't match other terminals that might start with "st".
         return true;
      }
      if (String_startsWith(termType, "screen") {
         return true;
      }
   }
   if (
      String_startsWith(termType, "xterm") ||
      String_eq(termType, "vt220") ||
      String_startsWith(termType, "tmux") ||
      String_eq(termType, "alacritty")
   ) {
      return true;
   }
   return false;
}
```

Reply:
Consider …

```c
   switch(termType[0]) {
      case 's':
         if (
            String_eq(…)
         ) {
            return true;
         }
         break;
      case 't':
         if (
            String_eq(…)
         ) {
            return true;
         }
         break;
      case 'x':
         if (
            String_eq(…)
         ) {
            return true;
         }
         break;
   }

   return false;
```

Reply:
@BenBE I would say it's okay either way.
Here's the suggested edit:

```c
static bool termSupportsDefinedKeys(const char* termType) {
   if (!termType)
      return false;

   switch (termType[0]) {
   case 'a':
      if (String_eq(termType, "alacritty"))
         return true;
      break;
   case 's':
      if (termType[1] == 't' && (termType[2] == '-' || !termType[2])) {
         // Suckless Terminal. Regex "st(-)?".
         // Don't match other terminals that might start with "st".
         return true;
      }
      if (String_startsWith(termType, "screen") {
         return true;
      }
      break;
   case 't':
      if (String_startsWith(termType, "tmux"))
         return true;
      break;
   case 'v':
      if (String_eq(termType, "vt220"))
         return true;
      break;
   case 'x':
      if (String_startsWith(termType, "xterm"))
         return true;
      break;
   }
   return false;
}
```